docs: Kubernetes deployment with the Helm chart (#969)
redpanda-data/documentation: docs/platform/kubernetes-deployment/kubernetes-tune-workers.mdx

---
title: Tuning Kubernetes Worker Nodes for Production
tags:
  - Helm
  - Helm deployment
---

<head>
    <meta name="title" content="Tuning Kubernetes Worker Nodes for Production | Redpanda Docs"/>
    <meta name="description" content="To get the best performance from your hardware, set Redpanda to production mode and run the auto-tuning tool. The auto-tuning tool identifies your hardware configuration and tunes itself to give you the best performance."/>
</head>

<!--import InstallRpkLinux from '../../partial_docs/_install-rpk-linux.mdx';-->

> To get the best performance from your hardware, set Redpanda to production mode and run the auto-tuning tool. The auto-tuning tool identifies your hardware configuration and optimizes the Linux kernel to give you the best performance.

Reviewer: Please highlight that the tuner must run on the host machines, not in the pods.

> To get the best performance from your hardware, set Redpanda to production mode and run the auto-tuning tool. The auto-tuning tool identifies your hardware configuration and optimizes the Linux kernel to give you the best performance.

Author: ```suggestion
To get the best performance from your hardware, set Redpanda to production mode on each worker node and run the auto-tuning tool. The auto-tuning tool identifies the hardware configuration on your worker node and optimizes the Linux kernel to give you the best performance.
```

## Prerequisites

Make sure that your current Linux user has root privileges. The auto-tuning tool requires privileged access to the Linux kernel settings.

## Install rpk

To run the auto-tuning tool, you need the Redpanda Keeper (rpk).

<!--<InstallRpkLinux/>-->

## Run the auto-tuning tool 

> Run the auto-tuning tool with root privileges.

Author: ```suggestion
Run the auto-tuning tool on each worker node that runs a Redpanda broker.
```

---
title: Tuning Kubernetes Worker Nodes for Production
tags:
  - Helm
  - Helm deployment
---

<head>
    <meta name="title" content="Tuning Kubernetes Worker Nodes for Production | Redpanda Docs"/>
    <meta name="description" content="To get the best performance from your hardware, set Redpanda to production mode and run the auto-tuning tool. The auto-tuning tool identifies your hardware configuration and tunes itself to give you the best performance."/>
</head>

<!--import InstallRpkLinux from '../../partial_docs/_install-rpk-linux.mdx';-->

> To get the best performance from your hardware, set Redpanda to production mode and run the [auto-tuning tool](../../introduction/autotune). The auto-tuning tool identifies your hardware configuration and optimizes the Linux kernel to give you the best performance.

Reviewer 2: What is the auto-tuning tool? Does this mean running `rpk redpanda tune all`? Also, @JakeSCahill, I recommend removing this link. I think that introduction/autotune page will be replaced by the new sizing guidelines.

Reviewer 2: Ah, now I see. I should have just kept reading! The Run the Auto-Tuning Tool info is really nice and should probably be in our main Redpanda tuning doc, since it's not specific to Kubernetes, right?

Author: This section? https://docs.redpanda.com/docs/platform/deployment/production-deployment/#set-redpanda-production-mode

Author: @shorgi is going to write a new doc page to address this content: https://github.com/redpanda-data/documentation/issues/979

> To get the best performance from your hardware, set Redpanda to production mode and run the [auto-tuning tool](../../introduction/autotune). The auto-tuning tool identifies your hardware configuration and optimizes the Linux kernel to give you the best performance.

Author: ```suggestion
To get the best performance from your hardware, set Redpanda to production mode and run the auto-tuning tool. The auto-tuning tool identifies your hardware configuration and optimizes the Linux kernel to give you the best performance.
```

  disk_irq               true     true     true                                       
  disk_nomerges          true     true     true                                       
  disk_scheduler         true     true     true                                       
  disk_write_cache       false    true     false      Disk write cache tuner is       
                                                      only supported in GCP           
  fstrim                 false    false    true                                       
  net                    true     true     true                                       
  swappiness             true     true     true                                       
  transparent_hugepages  false    false    true                                       
  ```

  </details>

Changes to the Linux kernel are not persisted. If a worker node restarts, make sure to run `sudo rpk redpanda tune all` on it again.

:::tip
You can use a privileged DaemonSet to schedule the auto-tuning tool on each worker node that runs a Redpanda broker.
:::

## Next steps

[Deploy the Redpanda cluster](../kubernetes-deploy).

> To learn more about what the auto-tuning tool does, see [Designed for Performance](../../introduction/autotune).

Reviewer 2: again, recommend removing this link and replacing with new sizing guidelines (when available)

Author: Will the sizing guidelines explain what the auto-tuner does? The current page is nice because it explains how Redpanda benefits from a tuned Linux kernel.

Author: See the conversation in the other PR: https://github.com/redpanda-data/documentation/pull/968#discussion_r1056185260

> To learn more about what the auto-tuning tool does, see [Designed for Performance](../../introduction/autotune).

Author: ```suggestion
```

  disk_nomerges          true     true     true                                       
  disk_scheduler         true     true     true                                       
  disk_write_cache       false    true     false      Disk write cache tuner is       
                                                      only supported in GCP           
  fstrim                 false    false    true                                       
  net                    true     true     true                                       
  swappiness             true     true     true                                       
  transparent_hugepages  false    false    true                                       
  ```

  </details>

Changes to the Linux kernel are not persisted. If a worker node restarts, make sure to run `sudo rpk redpanda tune all` on it again.

:::tip
You can use a privileged DaemonSet to schedule the auto-tuning tool on each worker node that runs a Redpanda broker.
:::

## Next steps

Deploy the cluster:

```bash
helm repo add redpanda https://charts.redpanda.com/
helm repo update
helm install redpanda redpanda/redpanda \
    --namespace redpanda \
    --create-namespace

> ```

Author: Yes, thank you! That was a leftover from the drafting stage. Will make that a link.

> Deploy the cluster:
> 
> ```bash
> helm repo add redpanda https://charts.redpanda.com/
> helm repo update
> helm install redpanda redpanda/redpanda \
>     --namespace redpanda \
>     --create-namespace
> ```

Author: ```suggestion
[Deploy the Redpanda cluster](../kubernetes-deploy).
```
